product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,510 @@
+---
+title: Performing a Postgres major version rolling upgrade 
+---
+
+## Performing a Postgres major version rolling upgrade
+
+This document outlines the procedure for performing a major version upgrade of a Postgres cluster using Postgres Distributed (PGD) as installed by Trusted Postgres Architect (TPA). The rolling upgrade process allows updating individual cluster nodes to a new major Postgres version while maintaining cluster availability and operational continuity. This approach minimizes downtime and ensures data integrity by allowing the rest of the cluster to remain operational as each node is upgraded sequentially. Upgrading your Postgres cluster to a newer major version is recommended to access improved features, performance enhancements, and security updates. This guide provides step-by-step instructions to facilitate a smooth and controlled upgrade process.
+
+!!! Note
+Ensure that your databases and configuration files are backed up before starting the upgrade process to help prevent data loss.
+!!!
+
+To perform the rolling upgrade of the Postgres major version on your PGD cluster, carry out the following steps on each node in the cluster:
+
+- Confirm the current Postgres version.
+- Check to see if the target node is the write leader of the cluster.
+- If it is the write leader, perform a planned switchover to make another node write leader so you can upgrade the target node and ensure cluster availability.
+- After confirming the target node is not the write leader, stop Postgres.
+- Install the new Postgres version and bdr_pg_upgrade.
+- Initialize the new Postgres version.
+- Copy the configuration for the old Postgres version to the configuration for the new Postgres version.
+- Rename the PGDATA directories.
+- Stop all Postgres instances.
+- Check that you can upgrade safely.
+- Run the Postgres major version upgrade.
+- Modify the Postgres service.
+- Start Postgres.
+- Validate the new Postgres version. 
+- Test the upgrade directly against the newly upgraded node.
+- Cleanup post-upgrade. 
+
+We will look at these steps in more detail below, specifically for upgrading a node from Postgres 15 to Postgres 16. 
+
+## Worked example
+
+A three-node cluster (kaftan, kaolin, and kaboom) we [deployed on AWS using TPA](https://www.enterprisedb.com/docs/pgd/latest/quickstart/quick_start_aws/) is currently running Postgres 15 on a Linux OS (Debian in this case). We want to upgrade a single node to Postgres 16 using `bdr_pg_upgrade`, and then upgrade the rest of the cluster similarly. This walkthrough focuses on the detailed steps for the target node, kaboom. 
```

Comment:
For this worked example, we start with a TPA cluster deployed using the AWS Quickstart. 

There are three nodes .....

We want to upgrade a single node to Postgres 16 using `bdr_pg_upgrade`, and then upgrade the rest of the cluster similarly. -> We want to upgrade a node to Postgres 16 as the first step of upgrading the entire cluster.

Reply:
Is it a TPA cluster or a PGD cluster though? With what I had written I was trying to convey that it was a PGD cluster deployed using TPA, isn't that accurate?

Reply:
A pgd cluster deployed with TPA